crates/sui-types/src/multisig.rs

```diff
@@ -260,15 +271,28 @@ impl MultiSig {
         })
     }
 
-    pub fn validate(&self) -> Result<(), FastCryptoError> {
+    pub fn init_and_validate(&mut self) -> Result<Self, FastCryptoError> {
         if self.sigs.len() > self.multisig_pk.pk_map.len()
             || self.sigs.is_empty()
             || self.bitmap > MAX_BITMAP_VALUE
         {
             return Err(FastCryptoError::InvalidInput);
         }
         self.multisig_pk.validate()?;
-        Ok(())
+        let sigs = std::mem::take(&mut self.sigs);
+        let mut new_sigs = Vec::with_capacity(sigs.len());
+
+        for s in sigs {
+            match s {
+                CompressedSignature::ZkLogin(mut z) => {
+                    z.inputs.init()?;
```

Comment:
open to suggestion here too - i need to initialize z.inputs with JWT details when parsed from_bytes